lib/config/global.ts

```diff
@@ -5,8 +5,10 @@ export class GlobalConfig {
   private static readonly OPTIONS: (keyof RepoGlobalConfig)[] = [
     'allowCustomCrateRegistries',
     'allowedPostUpgradeCommands',
+    'allowedPreUpgradeCommands',
```

Comment:
i think we should have one option for this.

eg `allowUpgradeCommands`

@rarkins @secustor @JamieMagee WDYT?

Reply:
I have tried to come up a scenario in which I would need to split this up but failed at it.   
On the other hand from a UX perspective I think it is better to mirror `pre` and `post` options.

Reply:
If I remove the `allowedPostUpgradeCommands` and `allowedPreUpgradeCommands` and add `allowedPeriUpgradeCommands` (for example), won't this cause backward compatibility issues (for those who already use `allowedPostUpgradeCommands`)?

Moreover, the commands executed in pre-update and in post-update have no interest in being the same. Post-update commands are mainly controls (ex: lint) while pre-update commands are mainly content generation commands (ex: xcodegen generates a directory required by cocoapods).

Reply:
i only mean the allow options, not the actual repo commands. we can migrate existing config options automatically.

so if we allow pre commands, i don't see a use case to not allow post commands, as both can nearly do anything.

Reply:
I think it would be best to have a unified "allowed commands" concept which applies to both pre and post

Reply:
What about my last commit: [refactor: unify allowPostUpgradeCommandTemplating/allowPreUpgradeComm…](https://github.com/renovatebot/renovate/pull/20423/commits/760a6999df54b9cd9cfee4727ad9c3ddb205e5ee) ?